Thanks for the deep review. Addressed in commit `a606f37`. Per-finding:

---

**blocker: pinning to an unmerged recipes-PR commit, not the PR HEAD** ✅

Repinned `github.com/vultisig/recipes` to `475ad1fe90db` (recipes#589 HEAD). All three CR-feedback commits are now picked up:
- `1d5f4ac9` — fix(cosmos): reject empty addresses in withdraw_rewards
- `e9919915` — fix(cosmos): reject empty validator addresses in redelegate
- `475ad1fe` — fix(cosmos): add delegator_address required-field check on redelegate

go.mod / go.sum delta is minimal — only the recipes pseudo-version bumps. `cosmossdk.io/math` and `github.com/cosmos/cosmos-sdk` move from indirect to direct, but that's `go mod tidy` mechanically responding to the test file's imports — not new transitive deps.

---

**preferably-blocking: load-bearing test has a false-positive failure mode** ✅

Tightened `TestRecipesSDKRegistersStakingDistributionInterfaces` with the marshal+unmarshal round-trip you suggested — extracted into a `roundTripAny` helper so both subtests use the same forced-codec-path pattern. The cached-pointer short-circuit in `cosmos-sdk/codec/types/interface_registry.go:UnpackAny` (line 343–346) is now bypassed.

**Verified the guard is real:** locally patched out `stakingtypes.RegisterInterfaces(ir)` and `distributiontypes.RegisterInterfaces(ir)` in `recipes/sdk/cosmos/sdk.go` (commented to `_` imports + commented call sites). The test now fails as expected:

```
--- FAIL: TestRecipesSDKRegistersStakingDistributionInterfaces/MsgBeginRedelegate_decodes_via_SDK_registry
    Error: no concrete type registered for type URL /cosmos.staking.v1beta1.MsgBeginRedelegate
           against interface *interface { ProtoMessage(); Reset(); String() string }

--- FAIL: TestRecipesSDKRegistersStakingDistributionInterfaces/MsgWithdrawDelegatorReward_decodes_via_SDK_registry
    Error: no concrete type registered for type URL /cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward
           against interface *interface { ProtoMessage(); Reset(); String() string }
```

Restored recipes/sdk.go after verification. The receipt-vs-claim gap is now closed.

---

**should-fix: PR body underdiscloses 56-commit recipes bump collateral** ✅

PR body now has a `## What this bumps beyond cosmos staking` section. Actual bump from verifier-main (`577976dfb292`) to recipes#589 HEAD (`475ad1fe90db`) is **59 commits, +4232 / −172 across 44 files**. Cosmos staking is 632 lines; remaining ~3600 lines disclosed line-by-line: AAVE v3, Polymarket CTF + Exchange, Cetus, Relay, THORChain + MayaChain + Uniswap v2 + ERC20 codegen rewrites, LiFi/swap-service refactors, RESOURCES.md.

On EVM indexer regression risk: this repo organizes indexers under `plugin/tx_indexer/pkg/chain/`, not `internal/indexer/{evm,thor}/`. The EVM indexer file (`plugin/tx_indexer/pkg/chain/evm.go`) has no dedicated test file in this repo — it's covered transitively via `plugin/tx_indexer/...` integration tests. Posted those receipts plus full-repo `go test ./...` in the PR body's `## ci receipts` section. None of the swap-aggregator codegen rewrites touch the verifier indexer's runtime path; they're consumed only via mcp-ts.

---

**should-fix: no Maya indexer test, no multi-msg test** ✅

Added three new tests:

1. `TestMayaChainIndexerRegistersStakingDistributionInterfaces` — same registry + hash assertions as the THORChain set, against `NewMayaChainIndexer` wired with `mayaChainTestSDK()` (`NewSDK(nil)` + `rtypes.RegisterInterfaces` + `RefreshCodec`).
2. `TestTHORChainIndexerDecodesMultiMsgWithStakingMsg` — `bank.MsgSend + staking.MsgBeginRedelegate` in one tx body, both decoded through the registry post-roundtrip.
3. `TestTHORChainIndexerDecodesCrossEcosystemMsgs` — your canary suggestion (see below).

---

**should-fix: weak test fixtures — addresses + pubKey + sigs are off-distribution** ✅

Replaced all three:
- Real cosmoshub-4 bech32 addresses: delegator `cosmos1n9qcqp4ds6c30zmnyxwq30vfd03cyuq3gpsllv` (QA-Stability-Vault), validators `cosmosvaloper1grgelyng2v6v3t8z87wu3sxgt9m5s03xfytvz7` and `cosmosvaloper1sjllsnramtg7ewxqwwrwjxfgc4n4ef9u2lcnj0`. Recipient `cosmos1ej2es5fjztqjcd4pwa0zyvaevtjd2y5w37wr9t` for the multi-msg test.
- Real on-curve secp256k1 keypair via `secp256k1.GenPrivKeyFromSecret([]byte("vultisig-verifier-cosmos-msg-types-test"))`. Deterministic across runs; pubkey bytes are 33-byte compressed with `0x02`/`0x03` prefix asserted in `testKeypair()`.
- Real signature via `priv.Sign(deterministic-msg)`. Returns 64 bytes R||S in lower-S form (the secp256k1 helper normalizes). Replaces zero-R `fakeSig`.

The earlier off-distribution fixtures relied on the SDK not currently validating; the new shapes will continue to work even if recipes adds structural validation upstream.

---

**should-fix: hash-shape test is comfort food, not a guard** ✅

Replaced `TestTHORChainIndexer_HashIsSignedShape` with `TestTHORChainIndexer_HashIsSignedTxSHA256`. New assertion is byte-exact equality:

```go
expected := expectedSignedHash(t, sdk, unsigned, sigs, pubKey)  // SHA256(Sign(unsigned)) upper-hex
assert.Equal(t, expected, hash)
```

`expectedSignedHash` calls `sdk.Sign()` independently and computes `SHA256` of the marshalled signed bytes — exactly the contract `ComputeTxHash` is supposed to honor. The earlier weak `!= SHA256(unsigned)` assertion is kept as a secondary cross-check (so the new test still flags a regression where the indexer hashes the unsigned bytes), but the load-bearing assertion is now the byte-equality one.

Also embedded the same byte-equality check into the two existing `_HashesMsgBeginRedelegate` / `_HashesMsgWithdrawDelegatorReward` tests so they no longer rely on shape-only properties.

---

**suggestion: pgerrcode direct/indirect shuffle is correct, but call it out cleaner** ⏭️

Skipped per your "Personal taste" framing. The bump is already large; splitting `pgerrcode` into a separate commit would mostly add bisect-noise without runtime benefit. If a follow-up cleanup PR is preferred, happy to spin one separately.

---

**q: backward-compat for THORChain/Maya msg parsing — canary test request** ✅

Added `TestTHORChainIndexerDecodesCrossEcosystemMsgs`: a tx body containing both:
- `rtypes.MsgDeposit` (THORChain, registered via `rtypes.RegisterInterfaces`)
- `stakingtypes.MsgBeginRedelegate` (generic cosmos, registered by `NewSDK`)

Both decoded through the same THORChain SDK's registry post-roundtrip; both included in the multi-msg `Sign + ComputeTxHash` path. This catches future registration shadowing between the two ecosystems if either set's registration gets dropped or replaced during `NewSDK` setup.

---

**verdict: REQUEST_CHANGES**

All three highest-priority items addressed:
1. ✅ Repinned to `475ad1fe`.
2. ✅ Test now actually exercises the registry path; verified via temp-removal experiment.
3. ✅ 56-commit collateral disclosed in PR body; full-repo + tx_indexer-suite receipts embedded.

Plus all four should-fix items closed (Maya mirror, multi-msg, real fixtures, real signed-shape guard) and the canary cross-ecosystem test added.

Sequence agreement: recipes#589 → this PR (with the merged-to-main commit re-pinned at that point) → agent-backend#225 → mcp-ts#72 → vultiagent-app#333. Will hold for recipes#589 to merge, then re-pin and request re-review.

— vultisig-ops

--- a/go.mod
+++ b/go.mod
@@ -34,7 +34,7 @@ require (
 	github.com/vultisig/commondata v0.0.0-20250710214228-61d9ed8f7778
 	github.com/vultisig/go-wrappers v0.0.0-20260116015747-e12e4d06cf57
 	github.com/vultisig/mobile-tss-lib v0.0.0-20250316003201-2e7e570a4a74
-	github.com/vultisig/recipes v0.0.0-20260430121755-475ad1fe90db
+	github.com/vultisig/recipes v0.0.0-20260503115421-d28b3880fd08
 	github.com/vultisig/vultiserver v0.0.0-20250825042420-c6e6ac281110
 	github.com/vultisig/vultisig-go v0.0.0-20260114092710-6c38516a0c85
 	github.com/xyield/xrpl-go v0.0.0-20230914223425-9abe75c05830

